staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -377,6 +385,35 @@ func (s *APIAggregator) PrepareRun() (preparedAPIAggregator, error) {
 		})
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.AggregatedDiscoveryEndpoint) {
+		s.discoveryAggregationController = NewDiscoveryManager(
+			aggregatorscheme.Codecs,
+			s.GenericAPIServer.Serializer,
+		)
+
+		// Inform discovery manager of all local api servers which contain
+		// crds/builtin types
+		s.discoveryAggregationController.AddLocalAPIService(
+			"kube-aggregator", s.GenericAPIServer.DiscoveryResourceManager)
+
+		i := 0
+		for delegate := s.GenericAPIServer.NextDelegate(); delegate != nil && delegate.NextDelegate() != nil; delegate = delegate.NextDelegate() {
+			s.discoveryAggregationController.AddLocalAPIService("delegate_"+strconv.Itoa(i), delegate.UnprotectedHandler())
+			i++
+		}
+
+		// Setup discovery endpoint
+		s.GenericAPIServer.Handler.GoRestfulContainer.Add(s.discoveryAggregationController.WebService())
```

Comment:
this surprises me a little.  didn't we just set `EnableAggregatedDiscoveryEndpoint=false` above?  Why do we expose thsi webservice?

Reply:
Similar to how today's discovery is implemented, that flag disables the /discovery/v1 functionality within the genericapiserver. The aggregator then sets up its own handler at /discovery/v1 to combine the response from every apiserver registered apiservice into one endpoint.